boot/freeldr/freeldr/lib/fs/fs.c

```diff
@@ -51,6 +51,9 @@ typedef struct tagDEVICE
 static FILEDATA FileData[MAX_FDS];
 static LIST_ENTRY DeviceListHead;
 
+#define IS_VALID_FILEID(FileId) \
+    ((FileId) < _countof(FileData) && FileData[(FileId)].FuncTable)
```

Comment:
I suggest adding an explicit cast to ULONG here, just in case someone does something stupid like
`LONG FileId = OpenBla(...);`

Reply:
@tkreuzer Please tell me whether the new code:
```c
#define IS_VALID_FILEID(FileId) \
    ((ULONG)(FileId) < _countof(FileData) && FileData[(ULONG)(FileId)].FuncTable)
```
is what you were thinking about.

Reply:
yes